sdk/tests/utils.test.ts

```diff
@@ -0,0 +1,127 @@
+import { assert } from 'chai'
+import { toPercentage, toSqrtPrice } from '../src/testUtils'
+import { PRICE_DENOMINATOR, calculatePriceAfterSlippage } from '../src/utils'
+
+describe('utils', () => {
+  describe('test calculatePriceAfterSlippage', () => {
+    it('no slippage up', async () => {
+      const price = toSqrtPrice(1n)
+      const slippage = toPercentage(0n)
+
+      const expected = PRICE_DENOMINATOR
+
+      const limitSqrt = calculatePriceAfterSlippage(price, slippage, true)
+      const limit = ((limitSqrt.v as bigint) * (limitSqrt.v as bigint)) / PRICE_DENOMINATOR
+
+      assert.equal(limit, expected)
+    })
+
+    it('no slippage down', async () => {
+      const price = toSqrtPrice(1n)
+      const slippage = toPercentage(0n)
+
+      const expected = PRICE_DENOMINATOR
+
+      const limitSqrt = calculatePriceAfterSlippage(price, slippage, false)
+      const limit = ((limitSqrt.v as bigint) * (limitSqrt.v as bigint)) / PRICE_DENOMINATOR
+
+      assert.equal(limit, expected)
+    })
+
+    it('slippage of 1% up', async () => {
+      const price = toSqrtPrice(1n)
+      const slippage = toPercentage(1n, 2n)
+
+      const expected = { v: 1009999999999821040955554n }
+
+      const limitSqrt = calculatePriceAfterSlippage(price, slippage, true)
+      const limit = ((limitSqrt.v as bigint) * (limitSqrt.v as bigint)) / PRICE_DENOMINATOR
+
+      assert.equal(limit, expected.v)
+    })
+
+    it('slippage of 1% down', async () => {
+      const price = toSqrtPrice(1n)
+      const slippage = toPercentage(1n, 2n)
+
+      const expected = { v: 990000000000756263920003n }
+
+      const limitSqrt = calculatePriceAfterSlippage(price, slippage, false)
+      const limit = ((limitSqrt.v as bigint) * (limitSqrt.v as bigint)) / PRICE_DENOMINATOR
+
+      assert.equal(limit, expected.v)
+    })
+
+    it('slippage of 0.5% up', async () => {
```

Comment:
Defining a function in a test as `async` seems unnecessary. There are no asycn function invocations here